pystack.py

```diff
@@ -0,0 +1,98 @@
+        self.load_metadata()
+
+    def list_files(self, directory, extensions=('*.yml', '*.yaml')):
+        _files = []
+        if not os.path.isdir(directory):
+            raise Exception("'{}' directory does not exist".format(directory))
+        else:
+            for ext in extensions:
+                _files.extend(glob.glob(os.path.join(directory, ext)))
+            if not _files:
+                raise Exception("Directory list is empty: .{}/".format(directory))
+            return _files
+
+    def read_yamlconfig(self, filepath):
+        with open(os.path.join(self.datadir, filepath), 'r') as f:
+            yamlconfig = yaml.load(f)
+        return yamlconfig
+
+    def load_metadata(self):
+        """Read metadata file, for example in data/pystack-meta.yaml"""
+        self.datafiles = self.list_files(self.datadir)
+        with open(os.path.join(self.datadir, self.metafile), 'r') as f:
+            metadata = yaml.load(f)
+            for service in metadata:
+                if metadata[service]['data'] in [os.path.basename(x) for x in self.datafiles]:
+                    # Initialize nested dict, such as os_config['nova']
+                    self.os_config[service] = {}
+                    # Set OpenStack service configuration, excluding redundant service name section by adding [service]
+                    self.os_config[service]['data'] = self.read_yamlconfig(metadata[service]['data'])[service]
+                    # Set service template file from metadata configuration
+                    self.os_config[service]['config'] = metadata[service]['config']
+
+    def load_servicedata(self):
+        """Read service.yaml file, for example in data/nova.yaml"""
+        self.datafiles = self.list_files(self.datadir)
+        with open(os.path.join(self.datadir, self.metafile), 'r') as f:
+            metadata = yaml.load(f)
+            for service in metadata:
+                if metadata[service]['data'] in [os.path.basename(x) for x in self.datafiles]:
+                    # Initialize nested dict, such as os_config['nova']
+                    self.os_config[service] = {}
+                    # Set OpenStack service configuration, excluding redundant service name section by adding [service]
+                    self.os_config[service]['data'] = self.read_yamlconfig(metadata[service]['data'])[service]
+                    # Set service template file from metadata configuration
+                    self.os_config[service]['config'] = metadata[service]['config']
+
+    def service_data(self, service):
+        """Returns service data from self.os_config"""
+        return json.dumps(self.os_config[service]['data'], indent=4, sort_keys=True)
+
+    def service_config(self, service):
+        """Returns service configuration files from self.os_config"""
+        return json.dumps(self.os_config[service]['config'], indent=4, sort_keys=True)
+
+    def __str__(self):
```

Comment:
awesome that you included this! Will make it easy to view the object however you want.  The only addendum is that you should also do a  __repr__ just to be clear.  the __repr__ can be the same exact code as this, though what I'd recommend is that you have your current __str__ become your __repr__ and have a new __str__ have even more formatting to it. (Maybe get fancy with [termcolor](https://pypi.python.org/pypi/termcolor)?

[A really good explanation of __str__ and __repr__](http://stackoverflow.com/questions/1436703/difference-between-str-and-repr-in-python)